src/controllers/cost-analysis/cost/analyze-costs.ts

```diff
@@ -0,0 +1,190 @@
+                        {
+                            key: 'date',
+                            name: 'date'
+                        }
+                    ],
+                    fields: [
+                        {
+                            name: 'values',
+                            operator: 'push',
+                            fields: [
+                                {
+                                    key: 'usd_cost',
+                                    name: 'usd_cost'
+                                }
+                            ]
+                        }
+                    ]
+                }
+            });
+
+            if (params.group_by) {
+                if (Array.isArray(params.group_by)) {
+                    for (const groupKey of params.group_by) {
+                        requestParams.query.aggregate[1].group.fields[0].fields.push({
+                            key: groupKey,
+                            name: groupKey
+                        });
+                    }
+                }
+            }
+
+            requestParams.query.aggregate.push({
+                sort: {
+                    key: 'date'
+                }
+            });
+
+        } else if (params.pivot_type === 'TABLE') {
+            requestParams.query.aggregate.push({
+                sort: {
+                    key: 'date'
+                }
+            });
+
+            requestParams.query.aggregate.push({
+                group: {
+                    keys: [],
+                    fields: [
+                        {
+                            name: 'values',
+                            operator: 'push',
+                            fields: [
+                                {
+                                    key: 'date',
+                                    name: 'date'
+                                },
+                                {
+                                    key: 'usd_cost',
+                                    name: 'usd_cost'
+                                }
+                            ]
+                        }
+                    ]
+                }
+            });
+
+            if (params.group_by) {
+                if (Array.isArray(params.group_by)) {
+                    for (const groupKey of params.group_by) {
+                        requestParams.query.aggregate[2].group.keys.push({
+                            key: groupKey,
+                            name: groupKey
+                        });
+                    }
+                }
+            }
+        }
+    }
+
+    if (params.filter) {
+        requestParams.query.filter = cloneDeep(params.filter);
+    }
+
+    requestParams.query.filter.push({
+        key: 'billed_at',
+        value: params.start,
+        operator: 'datetime_gte'
+    });
+
+    requestParams.query.filter.push({
+        key: 'billed_at',
+        value: params.end,
+        operator: 'datetime_lt'
+    });
```

Comment:
```suggestion
    if (params.filter) {
        requestParams.query.filter = cloneDeep(params.filter);
        
        requestParams.query.filter.push({
            key: 'billed_at',
            value: params.start,
            operator: 'datetime_gte'
        });
    
        requestParams.query.filter.push({
            key: 'billed_at',
            value: params.end,
            operator: 'datetime_lt'
        });
    }
```

엇 아래 두 구문이 if 블럭 안에 들어가야 할 것 같아요!